packages/@react-native-windows/cli/src/runWindows/runWindows.ts

```diff
@@ -54,16 +54,17 @@ function getPkgVersion(pkgName: string): string {
   return '';
 }
 
-enum RunWindowsPhase {
-  None,
-  AutoLink,
-  FindBuildTools,
-  NuGetRestore,
-  FindSolution,
-  Deploy,
-}
+/**
+ * Labels used by telemtry to represent current operation
+ */
+type RunWindowsPhase =
```

Comment:
what's the benefit of making these be strings rather than enums? is this just a style preference or is there more to it?

Reply:
A little bit stronger than a preference. TypeScript language designers have said on a few occasions that enums were a mistake. A recent mention of some of that here https://www.techatbloomberg.com/blog/10-insights-adopting-typescript-at-scale/

They have some sharp edges, but the bigger issue is interop and equivalency with JS, which doesn't really have an equivalent, and idomatically uses raw strings. This has some practical problems for actual interop, but is somewhat philosophical. Some style guides explicitly discourage enums for those reasons, and Typescript codebases I've worked in near universally use unions of strings instead of enums.

In this case specifically, we want to keep raw string values stable, so using raw strings can be a bit clearer. E.g. both Danny and I were tripped up by the enums thinking we were sending the numeric value (see his comments above). So, there's some more philosophical reasons to prefer unions, and it can make things clearer in this case.

They do still work well with dev tooling. E.g. Intellisense dropdowns work just like enums.

Reply:
Another great benefit of string unions is that they result in zero JS code generation, while enums add quite a bit of JS code. I.e. they are zero abstractions purely used by TypeScript compiler and completely disappearing after compilation and becoming just JS strings.

---
In reply to: [525410112](https://github.com/microsoft/react-native-windows/pull/6548#discussion_r525410112) [](ancestors = 525410112)